### ISSUE TITLE
Some different bebave on labels when `vmagent` relabel

### ISSUE THREAD

**Issue by dxtrzhang** [VictoriaMetrics/VictoriaMetrics#453]
**Describe the bug**
After using vmagent and vmalert to replace Prometheus, vmalert fires a lot of alerts which never be seen when we use Prometheus, and after switching back to Prometheus, no more alerts fired. So, we deep into the alerts that vmalert fired. We find some weird labels and all the metrics was collected from pushgateway.
```bash
curl http://localhost:9091/metrics
...
# TYPE no_instance untyped
no_instance{instance="",job="some_job",label="val1",test=""} 5555
# TYPE test_with_instance untyped
test_with_instance{instance="some_instance",job="some_job",label="val1",test=""} 5555
```
1. `honor_labels=true` and `instance` is null character:
Prometheus:
![image](https://user-images.githubusercontent.com/13900390/80572845-86557700-8a31-11ea-8a04-109bb5378088.png)
vmagent:
![image](https://user-images.githubusercontent.com/13900390/80573053-dfbda600-8a31-11ea-9752-e1f7626f5081.png)
vmagents drop all the labels that values null character, so whether instance is null character or no t be set can not be told, This causes vmagent use `__address__`'s value as instance.

2. `honor_labels=true` and `instance` is set:
Prometheus
![image](https://user-images.githubusercontent.com/13900390/80573443-886c0580-8a32-11ea-9d44-521f5156c963.png)
vmagent
![image](https://user-images.githubusercontent.com/13900390/80573477-9cb00280-8a32-11ea-972a-6ec0e8d87a12.png)

3. default port for scheme:
Prometheus
![image](https://user-images.githubusercontent.com/13900390/80573944-8191c280-8a33-11ea-959a-26320826d277.png)
vmagent
![image](https://user-images.githubusercontent.com/13900390/80573982-8f474800-8a33-11ea-959a-9bfa490379cd.png)

**Version**
```
$ ./victoria-metrics-prod --version
vmagent-20200415-095213-tags-v1.34.8-0-gccb887c0
```

**Others**
There are several reasons that cause these Inconsistents：
1. [vmagents drop all the labels that values null character](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/e3cc329d856670e6a53a9fbad2374fc775a43290/lib/protoparser/prometheus/parser.go#L233)
2. There some diff when relabel as this [blog](https://www.robustperception.io/life-of-a-label) mentioned in code comment

**Comment by valyala** [comment]
The issue must be fixed in both master  and cluster branches. The bugfix will be included in the next release. @dxtrzhang , thanks for the help!

**Comment by valyala** [comment]
@dxtrzhang , could you check whether `vmagent` built from the latest commit in `master` branch fixes all the issues mentioned in the original bug description?

**Comment by dxtrzhang** [comment]
> @dxtrzhang , could you check whether `vmagent` built from the latest commit in `master` branch fixes all the issues mentioned in the original bug description?

No Problem! I will check and update the result under this PR!

**Comment by dxtrzhang** [comment]
I've ran some tests, `vmagent` build from the `master` branch works fine now.

**Comment by valyala** [comment]
FYI, the bugfix has been included in the [release v1.35.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
